tools/upload_to_server.py

```diff
@@ -0,0 +1,232 @@
+        )
+
+    if api_key:
+        api.set_token(api_key)
+        if api.has_valid_token:
+            return api
+
+        if not username or not password:
+            raise click.BadParameter(
+                "API key is not valid."
+                " Provide valid API key, or username and password.",
+                param_hint="api-key"
+            )
+
+    api.login(username, password)
+    if api.has_valid_token:
+        return api
+    raise click.BadParameter(
+        "Invalid credentials."
+        " Provide valid API key, or username and password.",
+        param_hint="username & password"
+    )
+
+
+def create_installer(
+    api: ayon_api.ServerAPI, installer_info: InstallerInfo, force:bool
+):
+    server_installers: list[dict[str, Any]] = (
+        api.get_installers()["installers"])
+
+    matched_installer = None
+    for server_installer in server_installers:
+        platform_name: Union[str, None] = server_installer.get("platform")
+        version: Union[str, None] = server_installer.get("version")
+        if (
+            platform_name == installer_info.platform
+            and version == installer_info.version
+        ):
+            matched_installer = InstallerInfo(
+                version=server_installer.get("version"),
+                platform=server_installer.get("platform"),
+                filename=server_installer.get("filename"),
+                installer_path=installer_info.installer_path,
+                python_version=server_installer.get("pythonVersion"),
+                checksum=server_installer.get("checksum"),
+                checksum_algorithm=server_installer.get("checksumAlgorithm"),
+                size=server_installer.get("size"),
+                python_modules=server_installer.get("pythonModules"),
+                runtime_python_modules=(
+                    server_installer.get("runtimePythonModules")),
+            )
+            break
+
+    if matched_installer is not None:
+        if matched_installer == installer_info:
+            return False
+        if not force:
+            raise RuntimeError(
+                "Installer already exists on server"
+                " but with different values."
+                " Use --force to overwrite it."
+            )
+        print(
+            "Removing existing installer on server"
+            " because have different metadata"
+        )
+        api.delete_installer(matched_installer.filename)
+
+    print(
+        f"Creating installer {installer_info.version}"
+        f" on server {installer_info.filename}"
+    )
+    api.create_installer(
+        installer_info.filename,
+        installer_info.version,
+        installer_info.python_version,
+        installer_info.platform,
+        installer_info.python_modules.copy(),
+        installer_info.runtime_python_modules.copy(),
+        installer_info.checksum,
+        installer_info.checksum_algorithm,
+        installer_info.size,
+    )
+    return True
```

Comment:
Maybe it would be worth returning the installer_info instead of true or false? or a tuple such as (true, installer_info) if we created a new one or Fals if there was already one, which I take that's what True/False indicate here, right?